torch/csrc/jit/passes/quantization.cpp

```diff
@@ -18,8 +18,30 @@ void PropagateQuantInfo(std::shared_ptr<Graph>& graph) {
   throw std::runtime_error("Pass not implemented yet!");
 }
 
+static void addObserverFor(torch::jit::Value* v) {
+  torch::jit::Node* def = v->node();
+  torch::jit::Node* observer =
+      def->owningGraph()->create(at::Symbol::fromQualString("aten::observer"));
+  def->owningBlock()->appendNode(observer);
+  observer->moveAfter(def);
+  v->replaceAllUsesWith(observer->outputs()[0]);
+  observer->outputs()[0]->setUniqueName(v->uniqueName() + ".observed");
+  observer->addInput(v);
+}
+
 void InsertObserverNodes(std::shared_ptr<Graph>& graph) {
-  throw std::runtime_error("Pass not implemented yet!");
+  for (const auto& n : graph->nodes()) {
+    // Skip nodes that we've just added
+    if (n->kind().toQualString() == std::string("aten::observer")) {
```

Comment:
What's `aten::observer`? A grep shows no hits in our codebase

Reply:
You are correct, it doesn't exist yet.

This diff is a part of an effort to enable quantization. The given pass would insert so called observer nodes, which collect distribution of values taken by a given tensor. This data then will be used for computing scale and bias of quantization. The work on the observer operator itself is carried on in parallel (the name `aten::observer` is tentative).